src/Combinatorics/Matroids/properties.jl

```diff
@@ -0,0 +1,983 @@
+Note that `elem` needs to be in the complement of the `basis` in this case.
+
+
+# Example
+```jldoctest
+julia> M = fano_matroid();
+
+julia> fundamental_circuit(M, [1,2,4], 7)
+4-element Vector{Int64}:
+ 1
+ 2
+ 4
+ 7
+
+julia> fundamental_circuit(M, [1,2,4], 3)
+3-element Vector{Int64}:
+ 1
+ 2
+ 3
+```
+"""
+function fundamental_circuit(M::Matroid, basis::AbstractVector, elem::Union{IntegerUnion,Char,String})
+    if !(basis in bases(M))
+        error("The set is not a basis of M")
+    end
+    if !(elem in M.groundset)
+        error("The element is not in the groundset")
+    end
+    if elem in basis
+        error("The element is in the basis, but has to be in the complement")
+    end
+    return circuits(restriction(M,[elem; basis]))[1]
+end
+
+@doc Markdown.doc"""
+    fundamental_cocircuit(M::matroid, cobasis::Vector, elem::Union{IntegerUnion,Char,String})
+
+Return the unique circuit of the dual matroid of `M` in the union of the complement of `basis` and `elem`.
+See Section 2.1 of Oxl11 (@cite)
+Note that `elem` needs to be an element of the `basis` in this case.
+
+
+# Example
+```jldoctest
+julia> fundamental_cocircuit(fano_matroid(), [1,2,4], 4)
+4-element Vector{Int64}:
+ 4
+ 5
+ 6
+ 7
+```
+"""
+fundamental_cocircuit(M::Matroid, basis::AbstractVector, elem::Union{IntegerUnion,Char,String}) = fundamental_circuit(dual_matroid(M), setdiff(M.groundset, basis), elem)
+
+@doc Markdown.doc"""
+    spanning_sets(M::matroid)
+
+Return the list of spanning sets of the matroid `M`.
+These are all sets containing a basis.
+
+# Example
+```jldoctest
+julia> spanning_sets(uniform_matroid(2, 3))
+4-element Vector{Vector{Any}}:
+ [1, 3]
+ [2, 3]
+ [1, 2]
+ [1, 2, 3]
+```
+"""
+function spanning_sets(M::Matroid)
+    pm_bases = Vector{Set{Int}}(M.pm_matroid.BASES)
+    n = length(M.groundset)
+    gs = M.groundset
+    sets = Vector{Vector{Any}}()
+    for k in rank(M):n
+        for set in Oscar.Hecke.subsets(Vector(0:n-1),k)
+            for B in pm_bases
+                if issubset(B,set)
+                    push!(sets, [gs[i+1] for i in set])
+                    break
+                end
+            end
+        end
+    end
+    return sets
+end
+
+@doc Markdown.doc"""
+    independet_sets(M::matroid)
+
+Return the list of independent sets of the matroid `M`.
+These are all subsets of the bases.
+
+# Example
+```jldoctest
+julia> independent_sets(uniform_matroid(2, 3))
+7-element Vector{Vector{Any}}:
+ []
+ [1]
+ [2]
+ [3]
+ [1, 3]
+ [2, 3]
+ [1, 2]
+```
+"""
+function independent_sets(M::Matroid)
+    pm_bases = Vector{Set{Int}}(M.pm_matroid.BASES)
+    n = length(M.groundset)
+    gs = M.groundset
+    sets = [[]]
+    for k in 1:rank(M)
+        for set in Oscar.Hecke.subsets(Vector(0:n-1),k)
+            for B in pm_bases
+                if issubset(set,B)
+                    push!( sets, [gs[i+1] for i in set])
+                    break
+                end
+            end
+        end
+    end
+    return sets
```

Comment:
This function is almost identical with the previous `spanning_sets`. It should be refactored for less duplication.

Reply:
Done.